### PR TITLE
feat(resources): Use `resourcemanager/resources/armsubscriptions` for subscription resources

### LINKED ISSUE #9854: feat: Consider using Azure reourcemanager packages
### Which problem is this feature request solving?

There are some duplicate implementations present under [github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources](https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/resourcemanager/resources) that might provide more data that the `armXYZ` ones (tags for subscriptions, etc.)

### Describe the solution you'd like

1. Reach out to Azure to verify what packages are better to use & why
2. Decide on package by package basis whether we want to swap the implementation

### Pull request (optional)

- [ ] I can submit a pull request

### PR DESCRIPTION
**Pull request by disq**
Fixes #9854

### CODE REVIEW

File: plugins/source/azure/resources/services/subscription/locations.go [hunk 1]
```diff
@@ -14,16 +14,16 @@ func locations() *schema.Table {
 		Name:        "azure_subscription_subscription_locations",
 		Resolver:    fetchLocations,
 		Description: "https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations?tabs=HTTP#location",
```
Comment (disq) [inline plugins/source/azure/resources/services/subscription/locations.go:16]: Docs were already pointing to the right API responses so no change was made there.

File: website/tables/azure/azure_subscription_subscription_locations.md [hunk 2]
```diff
@@ -19,8 +19,9 @@ This table depends on [azure_subscription_subscriptions](azure_subscription_subs
 |_cq_id|UUID|
 |_cq_parent_id|UUID|
 |subscription_id|String|
+|metadata|JSON|
 |display_name|String|
 |id (PK)|String|
-|latitude|String|
```
Comment (disq) [inline website/tables/azure/azure_subscription_subscription_locations.md:24]: This is now a key under `metadata` along with `longitude` and bunch of new keys.

**Comment by candiduslynx** [comment]
IDK if we want to go through with this breaking change. @yevgenypats what's your take?

File: website/tables/azure/azure_subscription_subscription_locations.md (hunk 2 above)
Comment (hermanschaaf) [reply website/tables/azure/azure_subscription_subscription_locations.md:24]: Maybe we can keep these fields as aliases of the ones in the metadata JSON (using path  resolver) to keep this change backwards-compatible?

Comment (candiduslynx) [reply website/tables/azure/azure_subscription_subscription_locations.md:24]: > Maybe we can keep these fields as aliases of the ones in the metadata JSON (using path resolver) to keep this change backwards-compatible?

+1

Comment (erezrokah) [reply website/tables/azure/azure_subscription_subscription_locations.md]: This looks good, but +1 for making this backwards compatible

Comment (disq) [reply website/tables/azure/azure_subscription_subscription_locations.md]: Hopefully fixed in https://github.com/cloudquery/cloudquery/pull/10369/commits/6fc0ad846c5694398fd6b80d11ff8d3de1177a5a

**Review by hermanschaaf — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Pull request merged**